src/modules/connection.js

```diff
@@ -55,28 +57,38 @@ class Connection {
    * Since RabbitMQ advise to open one channel by process and node is mono-core, we keep only 1 channel for the whole connection.
    * @return {Promise} A promise that resolve with an amqp.node channel object
   */
-  getChannel() {
+  getChannel(queue, prefetch) {
     const url = this._config.host;
-    const { prefetch } = this._config;
     const connection = this.connections[url];
 
+    queue = queue || defaultChannel;
+    const channelToUse = prefetch ? queue : defaultChannel;
+
     // cache handling, if channel already opened, return it
-    if (connection && connection.chann) {
-      return Promise.resolve(connection.chann);
+    if (connection && connection.channels[channelToUse]) {
+      if (prefetch) {
+        throw new Error(`Channel already exist for queue ${queue}`);
```

Comment:
Why we are throwing here? if we asked for channel with same prefetch why don't we just return it?